### ISSUE TITLE
Translate site text

### ISSUE THREAD

**Issue by per1234** [arduino/reference-pl#2]
There is some text to translate that is not in this repository's content. Please translate the following list of words and add the translation as a reply here:

- Language Reference
- "Arduino programming language can be divided in three main parts: functions, values (variables and constants), and structure."
- Language
- functions
- variables
- structure
- Libraries
- Glossary
- Edit This Page
- `"The Arduino Reference text is licensed under a <a href=\"https://creativecommons.org/licenses/by-sa/3.0/\" target=\"_blank\" rel=\"noopener\">Creative Commons Attribution-Share Alike 3.0 License</a>."`
- `"Find anything that can be improved? <a href=\"https://github.com/arduino/reference-cz\" target=\"_blank\" rel=\"noopener\">Suggest corrections and new documentation via GitHub.</a>"`
- `"Doubts on how to use Github? Learn everything you need to know in <a href=\"https://create.arduino.cc/projecthub/Arduino_Genuino/contribute-to-the-arduino-reference-af7c37\" target=\"_blank\" rel=\"noopener\">this tutorial</a>."`

**Comment by tomb0x** [comment]
- Omówienie Języka (this is my proposition)
- "Język programowania Arduino można podzielić na trzy główne części: funkcje, wartości (zmienne i stałe) oraz strukturę."
- Język
- funkcje
- zmienne
- struktura
- Biblioteki
- Słowniczek
- Redaguj tą stronę
- "Tekst Omówienia Arduino jest licencjonowany w ramach <a href=\"https://creativecommons.org/licenses/by-sa/3.0/\" target=\"_blank\" rel=\"noopener\">Licencji Creative Commons Attribution-Share Alike 3.0</a>."
-    "Szukasz czegoś co można ulepszyć? <a href=\"https://github.com/arduino/reference-cz\" target=\"_blank\" rel=\"noopener\">Zaproponuj poprawki i nową dokumentację za pomocą GitHub.</a>"
-  "Wątpliwości, jak używać Github? Dowiedz się wszystkiego, co musisz wiedzieć w <a href=\"https://create.arduino.cc/projecthub/Arduino_Genuino/contribute-to-the-arduino-reference-af7c37\" target=\"_blank\" rel=\"noopener\">tym samouczku</a>."

**Comment by tomb0x** [comment]
I translated some text in https://github.com/tomb0x/reference-pl-tmp ...
Can this be combined with the https://github.com/arduino/reference-pl afterwards?
I'm just starting with git / github ...

**Comment by per1234** [comment]
Thanks so much for the great work @tomb0x! I'm just waiting on my pull request to get the arduino.cc infrastructure set up to be merged and then I'll transfer these translations over.

>I translated some text in https://github.com/tomb0x/reference-pl-tmp ...
Can this be combined with the https://github.com/arduino/reference-pl afterwards?

It looks like you're making some excellent progress already! Yes, you can easily submit those translations as pull requests. I'm planning to provide you with some instructions on how to proceed. I'm just waiting for one of my colleagues from Arduino to merge my pull request to add the English language content to this repository first (https://github.com/arduino/reference-pl/pull/1) so we can get started.

**Comment by tomb0x** [comment]
>     * Omówienie Języka (this is my proposition)
[Language Reference]: I change the "Omówienie Języka" to "Studium Języka". I have already changed in https://github.com/tomb0x/reference-pl-tmp

**Comment by tomb0x** [comment]
which of the 'special texts' (like ["TAGS "] or "tags" in .adoc can be translated?
- (1) in German raw.githubusercontent.com/arduino/reference-de/master/Language/Variables/Constants/constants.adoc "Constants" => "Konstanten" but in the Italian translation raw.githubusercontent. com/arduino/reference-it/master/Language/Variables/Constants/constants.adoc "Constants" is not translated. So, do these strings in quotation marks ["Variables"], ["Constants"], etc are the names of the folders in the reference tree or not? Translate them (like in German) or not (like in Italian)?
- (2): near the end of the file raw.githubusercontent.com/arduino/reference-de/master/Language/Functions/Digital%20IO/digitalRead.adoc -> is # BEISPIEL # (instead of # EXAMPLE #) - should I translate it? (in Polish translation not German :-) )
I also translated it in (for example) raw.githubusercontent.com/tomb0x/reference-pl-tmp/master/Language/Functions/Digital%20IO/digitalRead.adoc -> (#PRZYKŁAD#), but as I read raw.githubusercontent.com/arduino/reference-en/master/AsciiDoc_sample/Reference_Terms/AsciiDoc_Template-Parent_Of_Entities.adoc -> "please add the tag #DEFINITION #...", "please add the tag #EXAMPLE#" etc -> I do not know whether it is added exactly as it was written (in English) or in the translated version?

example: which words can be translated?
---
title: digitalRead()
categories: [ "Functions" ]
subCategories: [ "Digital I/O" ]
---
title? categories? subCategories? "Functions"? or "Digital I/O"?

**Comment by per1234** [comment]
>So, do these strings in quotation marks ["Variables"], ["Constants"], etc are the names of the folders in the reference tree or not?

They must match the appropriate subcategory name shown on the Language Reference home page:
https://www.arduino.cc/reference/de
You defined the subcategory names in your translations of this list:
https://github.com/arduino/reference-pl/issues/3
so in the case of the constants reference page it should be translated to:
```
subCategories: [ "Konwersja" ]
```

>Translate them (like in German) or not (like in Italian)?

The Italian reference is not a great choice for a model to follow because it's still in the early stages of translation. If you look at the Italian reference home page, you'll notice there is no link to the constants reference page because this line was not translated yet:
https://www.arduino.cc/reference/it/

>is # BEISPIEL # (instead of # EXAMPLE #) - should I translate it?

Yes, translate it. This is what that is for:
![Clipboard01](https://user-images.githubusercontent.com/8572152/59060126-456ed580-8855-11e9-8343-ebf4e94ce936.png)

>which words can be translated?
>title: digitalRead()

Yes. This determines the text of the link on the Language Reference home page as well as the title of the reference page.

In this case there is no need to make a change because `digitalRead()` is the same in every language.

>categories: [ "Functions" ]

Yes. This doesn't appear to be used for anything currently, but it's probably still a good idea to translate it. It should match the translations you made in this issue [above](https://github.com/arduino/reference-pl/issues/2#issuecomment-496935998). In this case it would be:
```
categories: [ "funkcje" ]
```

>subCategories: [ "Digital I/O" ]

Yes. This determines which subcategory the link will appear under on the Language Reference home page and is also shown under the title on the reference page.

**Comment by tomb0x** [comment]
what about slashes in folder names? in reference-en is (just for example): subCategories [ "Analog I/O" ] but folder name is Analog IO (without slash). In Polish translation, I have translated to "Analogowe We-Wy" in the folder name and "Analogowe We/Wy" into information about the subcategory in the file... Will this cause any (potential) problems?

**Comment by per1234** [comment]
I don't know if it's a good idea to translate the folder or file names. None of the other translation repositories have done this. Translating the folder and file names will make it more difficult to sync between repositories. Using the same folder and file names makes it so people can easily switch between the English and translated versions of the reference content by only changing the language code in the URL.

The primary benefit of translating the folder and filenames is this would cause the breadcrumbs to be translated.

**Comment by tomb0x** [comment]
you're right, thanks

**Comment by per1234** [comment]
The English language content has now been added to the repository so you can now start submitting pull requests for your translations. I can provide some detailed instructions for this process, but I'd like to know whether you prefer to use the GitHub website interface exclusively, are you using Git client software on your computer (e.g., GitHub Desktop), or do you use Git from the command line? I can tailor the instructions accordingly.

**Comment by tomb0x** [comment]
how can I transfer all my work from tomb0x/reference-pl-tmp to arduino/reference-pl using github (not individual files but all at once)?

**Comment by per1234** [comment]
I haven't found a way to submit pull requests directly from your reference-pl-tmp repository to this repository using the GitHub web interface. You can only do that from a fork. It's also a bit difficult to copy over 100 files to a repository using the GitHub web interface so I submitted a pull request to your fork adding the files:
https://github.com/tomb0x/reference-pl/pull/1
You can click the "Merge" button on that pull request to bring those files into your fork.

After that, you can submit a pull request from your fork to this repository to add your translations by clicking the "New pull request" button and then the "Create pull request" button.

**Comment by tomb0x** [comment]
language polishing: (Language Reference) 'Dokumentacja języka' (not 'Studium Języka'...) like in readme.adoc. IMO it's easier to understand for Polish language people.

**Comment by tomb0x** [comment]
one more change:
"Tekst Omówienia Arduino jest licencjonowany w ramach <a href="https://creativecommons.org/licenses/by-sa/3.0/\" target="_blank" rel="noopener">Licencji Creative Commons Attribution-Share Alike 3.0."
to
"Treść dokumentacji Arduino jest na licencji  <a href="https://creativecommons.org/licenses/by-sa/3.0/\" target="_blank" rel="noopener">Licencji Creative Commons Attribution-Share Alike 3.0."

**Comment by per1234** [comment]
OK. I have submitted a pull request to make the changes you requested here and in https://github.com/arduino/reference-pl/issues/3#issuecomment-502105176.
